crates/accelerate/src/permutation.rs

```diff
@@ -111,10 +196,31 @@ fn _get_ordered_swap(py: Python, permutation_in: PyArrayLike1<i64>) -> PyResult<
     Ok(get_ordered_swap(&view).to_object(py))
 }
 
+/// Decompose a SWAP pattern into a series of SWAP gate indices to implement them.
+/// For example, let the pattern be [1, 2, 3, 4, 0, 6, 5], which contains the two cycles
+/// [1, 2, 3, 4, 0] and [6, 5]. These can then be implemented with the SWAPs
+/// [(0, 3), (1, 2), (0, 4), (1, 3)], respectively [(6, 5)].
+/// If ``invert`` is True, reverse the indices before computing the SWAPs.
+#[pyfunction]
+#[pyo3(signature = (pattern, invert_order=true))]
+fn _pattern_to_swaps(
```

Comment:
why this function is needed in addition to `_synth_permutation_acg` ?

Reply:
Removed in 0743d23 🙂